docs/rules/remove-redundant-type-cast.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Remove Redundant Type Casts
+ruleId: RemoveRedundantTypeCast
+since: 3.14.0
+minJavaVersion: 1.1
+remediationCost: 1
+links:
+description:
+    All method invocations of toString() on a String element are not needed. Applying this rule will remove such method calls.\n\nFor example, '"string".toString()' becomes '"string"'.\n\nThis transformation improves performance. RemoveToStringOnStringRule_name=Remove toString() on String.
+
+tags: ["Java 1.1", "Readability", "Performance"]
+---
+
+# Remove Redundant Type Casts
+
+[[toc]]
+
+## Properties
+
+<RuleProperties />
+
+
+## Description
+
+The rule first searches the next type-cast operation. If the expression is casted to a type which already is exactly the type of the expression, then the type casting prefix is removed. Additionally, also parentheses involved in the cast operation will be removed if they are not necessary any more. This rule regards two types as exactly the same only when both types have also exactly the same generic arguments.
+
+## Benefits
+
+This rule increases readability of code and additionally it improves performance by removing unnecessary cast operations.
```

Comment:
Performance is not actually a topic for this rule. The java compiler would also remove the redundant casting.